src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.logging.Parser;
+import duke.logging.Storage;
+import duke.logging.TaskList;
+import duke.logging.Ui;
+
+/**
+ * A personal chat bot that helps a person keep track of various things.
+ */
+public class Duke {
+    private final Storage storage;
+    private TaskList tasks = new TaskList(new ArrayList<>());
+    private final Ui ui;
```

Comment:
### Should `storage` and `ui` be in uppercase?

Honestly, I wouldn't consider this to be violations, but I'll leave it to you to judge: 
    - Your variables are final whenever possible, which is good, 
    - However, that means it probably should be written in uppercase according to the coding convention. 
    - My personal belief is that only static final variables should be written in uppercase, but the coding convention isn't clear on that. 
    - However, an example they provided was `final int COLOR_RED = 1;` which leans towards the former,

### I would leave them as non-final to avoid meeting this problem.